src/tests/krate.rs

```diff
@@ -414,6 +414,69 @@ fn exact_match_on_queries_with_sort() {
     assert_eq!(json.crates[3].name, "other_sort");
 }
 
+#[test]
+fn loose_search_order() {
+    let (app, anon, user) = TestApp::init().with_user();
+    let user = user.as_model();
+
+    let ordered = app.db(|conn| {
+        // exact match should be first
+        let one = CrateBuilder::new("temp", user.id)
+            .readme("readme")
+            .description("description")
+            .keyword("kw1")
+            .expect_build(conn);
+        // file shouldn't match at all
+        CrateBuilder::new("file", user.id)
+            .readme("readme")
+            .description("description")
+            .keyword("kw1")
+            .expect_build(conn);
+        // temp_udp should match second
+        let two = CrateBuilder::new("temp_utp", user.id)
+            .readme("readme")
+            .description("description")
+            .keyword("kw1")
+            .expect_build(conn);
+        // evalrs should match 3rd because of readme
+        let three = CrateBuilder::new("evalrs", user.id)
+            .readme(
+                r#"$ echo 'println!("Hello World!")' | evalrs
+   Compiling evalrs_temp v0.0.0 (file:///tmp/evalrs_temp.daiPxHtjV2VR)
+    Finished debug [unoptimized + debuginfo] target(s) in 0.51 secs
+     Running `target\debug\evalrs_temp.exe`
+Hello World!"#,
+            )
+            .description("description")
+            .keyword("kw1")
+            .expect_build(conn);
+        // tempfile should appear 4th
+        let four = CrateBuilder::new("tempfile", user.id)
+            .readme("readme")
+            .description("description")
+            .keyword("kw1")
+            .expect_build(conn);
+        // mkstemp should appear 5th
```

Comment:
Can you put a bit more context in here? Why should this be 5th instead of tempfile?

Reply:
The only reason that is the case is because it was inserted later. I can remove that one.